rules/CodeQuality/Rector/MethodCall/AssertCompareOnCountableWithMethodToAssertCountRector.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\PHPUnit\CodeQuality\Rector\MethodCall;
+
+use Countable;
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\StaticCall;
+use PhpParser\Node\Identifier;
+use PHPStan\Type\ObjectType;
+use Rector\PHPUnit\NodeAnalyzer\TestsNodeAnalyzer;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\PHPUnit\Tests\CodeQuality\Rector\MethodCall\AssertCompareOnCountableWithMethodToAssertCountRector\AssertCompareOnCountableWithMethodToAssertCountRectorTest
+ */
+class AssertCompareOnCountableWithMethodToAssertCountRector extends AbstractRector
+{
+    public function __construct(
+        private readonly TestsNodeAnalyzer $testsNodeAnalyzer
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+$this->assertSame(1, $countable->count());
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+$this->assertCount(1, $countable);
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<MethodCall|StaticCall>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class, StaticCall::class];
+    }
+
+    /**
+     * @param MethodCall|StaticCall $node
+     */
+    public function refactor(Node $node): MethodCall|StaticCall|null
+    {
+        if (! $this->testsNodeAnalyzer->isPHPUnitMethodCallNames($node, ['assertSame', 'assertEquals'])) {
+            return null;
+        }
+
+        if ($node->isFirstClassCallable()) {
+            return null;
+        }
+
+        if (count($node->getArgs()) < 2) {
+            return null;
+        }
+
+        $right = $node->getArgs()[1]
+->value;
+
+        if (
+            ($right instanceof MethodCall)
+            && $right->name instanceof Identifier
+            && $right->name->toLowerString() === 'count'
+            && $right->getArgs() === []
+        ) {
+            $type = $this->getType($right->var);
+
+            if ((new ObjectType(Countable::class))->isSuperTypeOf($type)->yes()) {
```

Comment:
use normal `"Countable"` string, even usually native php class not prefixed by php scoper, just to avoid issue on downgrade:

```suggestion
            if ((new ObjectType('Countable'))->isSuperTypeOf($type)->yes()) {
```